src/plugins/crypto/CMakeLists.txt

```diff
@@ -5,83 +5,33 @@ include (LibAddPlugin)
 #
 # in order to set up OpenSSL development files manually, please provide:
 # - OPENSSL_INCLUDE_DIR
-# - OPENSSL_LIBRARIES
+# - OPENSSL_CRYPTO_LIBRARY
 #
 # in order to set up libgcrypt development files manually, please provide:
-# - LIBGCRYPT_INCLUDE_DIR
-# - LIBGCRYPT_LIBRARIES
+# - GCRYPT_INCLUDE_DIRS
+# - GCRYPT_LIBRARIES
 #
-if (DEPENDENCY_PHASE)
-	#
-	# HACK for Apple
-	# check if library detection is required
-	#
-	if (APPLE AND DEFINED OPENSSL_INCLUDE_DIR AND DEFINED OPENSSL_LIBRARIES)
-		set (OPENSSL_MANUAL_OSX_HACK on)
-		set (OPENSSL_FOUND on)
-	else ()
-		set (OPENSSL_MANUAL_OSX_HACK off)
-	endif ()
-
-	if (APPLE AND DEFINED LIBGCRYPT_INCLUDE_DIR AND DEFINED LIBGCRYPT_LIBRARIES)
-		set (LIBGCRYPT_MANUAL_OSX_HACK on)
-		set (LIBGCRYPT_FOUND on)
-	else ()
-		set (LIBGCRYPT_MANUAL_OSX_HACK off)
-	endif ()
 
-	set (plugin crypto_openssl)
-	if (NOT OPENSSL_MANUAL_OSX_HACK)
-		include (FindPkgConfig)
-		pkg_search_module (OPENSSL openssl)
-	endif ()
-
-	if ((NOT OPENSSL_FOUND) AND (NOT OPENSSL_MANUAL_OSX_HACK))
-		remove_plugin ( ${plugin} "OpenSSL development files not found")
-	else  ()
+if(APPLE) # gi
+	set(OPENSSL_ROOT_DIR "/usr/local/opt/openssl")
+endif(APPLE)
 
-		try_compile (HAS_OPENSSL_4SURE
-			"${CMAKE_BINARY_DIR}"
-			"${PROJECT_SOURCE_DIR}/src/plugins/crypto/compile_openssl.c"
-			CMAKE_FLAGS
-				-DINCLUDE_DIRECTORIES:STRING=${OPENSSL_INCLUDE_DIR}
-				-DLINK_LIBRARIES:PATH=${OPENSSL_LIBRARIES}
-		)
+find_package(OpenSSL)
```

Comment:
It does not matter if the options are reset, the add_plugin remembers the arguments it got.

find_package should only be used in the DEPENDENCY_PHASE.

I think the OPENSSL_MANUAL_OSX_HACK already does what you are trying to achieve.

And as said: the crypto plugin has no priority for this PR, its better if you revert these changes.